movai_core_shared/core/zmq_server.py

```diff
@@ -95,20 +92,19 @@ async def _handle(self, msg_buffer) -> None:
             if response.get("response") is None:
                 response = {"response": response}
             await self.handle_response(response)
-            index = len(msg_buffer) - 1
-            msg_buffer[index] = json.dumps(response).encode("utf8")
-            with self._lock:
-                self._socket.send_multipart(msg_buffer)
+            buffer[index] = json.dumps(response).encode("utf8")
+            self._socket.send_multipart(buffer)
             if self._debug:
                 self._logger.debug(f"{self._name} successfully sent a respone.")
 
     def close(self) -> None:
         """close the zmq socket."""
-        self._socket.close()
-        self._ctx.destroy()
-        self._socket = None
-        self._ctx = None
-        self._initialized = False
+        if self._initialized:
+            self._socket.close()
+            self._ctx.destroy()
```

Comment:
I remember having issues using destroy, can you use term instead ?

Reply:
Well I have tested it with this and I prefer not to change it now...